rfcs/0013-deprecate-currency-factory.md

```diff
@@ -0,0 +1,210 @@
+  
+### Remove Asset ID Reservation
+
+> We do not need [asset ID reservation]. We just runtime config assets to be 
+permissioned. When we will have permisionless assets they may already have 
+reservation (any way making fork with reservation is easy later).
+
+Asset reservation helps us automate new asset IDs. However, for the scope of 
+release two - this may not be necessary. We could instead hard-code asset IDs 
+for LP tokens and other instances.
+
+Given we can currently manage without automated asset creation, as we can forgo
+the asset creation requirements for release two and ensure that all asset data
+is still made available.
+
+#### Consequences
+
+* Minor runtime migration will be needed for deleting Currency Factory's 
+storage.
+
+### Use an Externally Maintained Asset ID Reservation System
+
+Neither option presented seems to handle the reserving of asset IDs. Therefore, 
+neither of these will fully replace currency factory and would more correctly 
+replace our own pallet-assets. A replacement for pallet-assets may be a good 
+idea but would require much more substantial changes.
+
+* **Parity's pallet-assets**
+  
+  Forking this pallet and maintaining extra features we need (hold) is an 
+  option.
+  
+  > It is not clear from RFC why substrate assets are good. IMHO it should be 
+  clarified - year of man work to make UI/QA/FE/users happy is what we lack. 
+  IMHO better to have fork of assets and collaborate with ecosystem to tune 
+  wallets. Do not write own wallets.
+  
+  #### Pros
+  
+  * Very nice UI compatibility with Polkadot Dashboard
+
+  * Maintained by Parity
+
+  * Simple Interface
+
+  * Can run two pallets at same time for two purposes (external and mintable) 
+  (Demonstrated by Moonbeam)
+
+  * Handles WASM contracts addresses as IDs
+
+  * Permissionless when needed
+
+  * Supported by Polkadot wallets
+  
+  * Supports asset metadata
+  
+  #### Cons
+  
+  * No asset ID reservation system, only prevents duplicates
+
+  * Hold functionality is not present (We depend on this in several cases)
+
+* **Moonbeam's pallet-asset-manager**
+
+Note: Not looking at using or forking this directly, but instead how they use 
+different instances of the same pallet for mintable and external assets.
+
+  #### Pros
+  
+  * Maintained by moonbeam
+  
+  #### Cons
+  
+  * Partially specialized
+
+  * No asset ID reservation system, only prevents duplicates
+
+#### Consequences
+
+* Any of these options will require a difficult runtime migration as we move 
+existing storage for pallet-assets/pallet-asset-registry/pallet-currency-factory 
+to an alternative.
+
+## Conclusion
+
+After analyzing possible solutions, the best course of action to deprecate 
+Currency Factory is to maintain a fork of Parity's pallet-assets, embrace the 
+design Moonbeam uses to maintain two instances of their pallet, and to remove 
+our asset ID reservation system (for release two).
```

Comment:
what benefits does forking bring over simply wrapping the pallet?

Reply:
Could we upstream the changes we need? This may be a more long-term task, so possibly out of scope for what we want to do with this RFC; perhaps a "future possibilities" section would be good?

Reply:
Up-streaming is an option, probably not one for release two tho.

Reply:
makes sense, i think that should be mentioned as explicitly out of scope then